docs/types/value-types.rst

```diff
@@ -813,6 +813,12 @@ functions.
     Marking the parameters as ``calldata`` only affects the implementation of the external function and is
     meaningless in a function pointer on the caller's side.
 
+.. warning::
+    Comparison of internal function pointers can have unexpected results with the legacy optimizer enabled,
+    as it can collapse identical functions into one, which will then lead to said functions comparing as equal instead of not.
```

Comment:
```suggestion
    as it can collapse identical functions into one, which will then lead to said function pointers comparing as equal instead of different.
```
I think that you meant the functions pointers are going to be equal (since the functions were already collapsed).
Also, feels a little bit weird reading `instead of not`, maybe it's better to just say `different`.